src/Release/CreateRelease.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Clearbooks\Labs\Release;
+use Clearbooks\Labs\Release\Gateway\ReleaseGateway;
+use Clearbooks\Labs\Release\UseCase\CreateRelease\Request;
+use Clearbooks\Labs\Release\UseCase\CreateRelease\Response;
+use Clearbooks\Labs\Release\CreateRelease\ResponseModel;
+
+/**
+ * @author: Ryan Wood <ryanw@clearbooks.co.uk>
+ * @created: 15/07/15
+ */
+class CreateRelease
+{
+    /**
+     * @var ReleaseGateway
+     */
+    private $releaseGateway;
+
+    /**
+     * @param ReleaseGateway $releaseGateway
+     */
+    function __construct( ReleaseGateway $releaseGateway )
+    {
+        $this->releaseGateway = $releaseGateway;
+    }
+
+
+    /**
+     * @param Request $request
+     * @return Response
+     */
+    public function execute( Request $request )
+    {
+        $response = $this->getResponse( $request );
+
+        if( !$response->isSuccessful() ) {
+            return $response;
+        }
+
+        $response->setId( $this->releaseGateway->addRelease( $request->getReleaseName(), $request->getReleaseInfoUrl() ) );
+
+        return $response;
+    }
+
+    /**
+     * @param Request $request
+     * @return ResponseModel $response
+     */
+    private function getResponse( Request $request )
```

Comment:
I think this name is somewhat misleading.

It should be ```validateRequestAndGetResponse```

And as soon as you name it that, you know it has too many responsibilities.